### PR TITLE
Added changes for custom url for self hosted data hub instances

### PR DESCRIPTION
**Pull request by sackh**
Signed-off-by: Kharude, Sachin <sachin.kharude@here.com>

This PR includes changes for private instances of DataHub APIs. 
With this change, the user can provide a different base URL for self-hosted Data Hub APIs.

### CODE REVIEW

File: xyzspaces/spaces.py [hunk 1]
```diff
@@ -65,9 +66,15 @@ class Space:
     """
 
     @classmethod
-    def from_id(cls, space_id: str) -> "Space":
-        """Instantiate a space object for an existing space ID."""
-        api = HubApi()
+    def from_id(cls, space_id: str, server: str = XYZ_BASE_URL,) -> "Space":
```
Comment (deeplook) [inline xyzspaces/spaces.py:69]: Remove suspicious trailing comma after XYZ_BASE_URL?

File: tests/hub/test_misc.py [hunk 2]
```diff
@@ -176,7 +176,7 @@ def test_get_space_tile_sampling(api):
     len_viz_low = len(tile_viz_low["features"])
     len_viz_med = len(tile_viz_med["features"])
     len_viz_high = len(tile_viz_high["features"])
-    assert len_raw > len_viz_off > len_viz_low > len_viz_med >= len_viz_high
+    assert len_raw > len_viz_off >= len_viz_low > len_viz_med >= len_viz_high
```
Comment (deeplook) [inline tests/hub/test_misc.py:179]: Is this related to the topic of this PR?

File: tests/space/test_space_objects.py [hunk 3]
```diff
@@ -876,4 +876,4 @@ def test_get_space_tile_sampling(api):
     len_viz_low = len(tile_viz_low)
     len_viz_med = len(tile_viz_med)
     len_viz_high = len(tile_viz_high)
-    assert len_raw > len_viz_off > len_viz_low > len_viz_med >= len_viz_high
+    assert len_raw > len_viz_off >= len_viz_low > len_viz_med >= len_viz_high
```
Comment (deeplook) [inline tests/space/test_space_objects.py:879]: Same here, related to the topic of this PR?

**Comment by deeplook** [comment]
Please add a proclamation entry, too!

File: tests/hub/test_misc.py (hunk 2 above)
Comment (sackh) [reply tests/hub/test_misc.py:179]: No, when I pushed the changes this test was failing hence, I fixed it along with my changes in a separate commit.

File: tests/space/test_space_objects.py (hunk 3 above)
Comment (sackh) [reply tests/space/test_space_objects.py:879]: No, when I pushed the changes this test was failing hence, I fixed it along with my changes in a separate commit.

File: xyzspaces/spaces.py (hunk 1 above)
Comment (sackh) [reply xyzspaces/spaces.py]: fixed.

**Comment by sackh** [comment]
Added proclamation entry.

**Review by deeplook — APPROVED** [review]
